src/java.base/share/classes/sun/util/locale/LanguageTag.java

```diff
@@ -221,127 +220,111 @@ public static LanguageTag parse(String languageTag, ParsePosition pp,
             throw new IllformedLocaleException(errorMsg.toString(), pp.getErrorIndex());
         }
 
-        return tag;
+        return new LanguageTag(language, script, region, privateuse, extlangs, variants, extensions);
     }
 
     //
     // Language subtag parsers
     //
 
-    private boolean parseLanguage(StringTokenIterator itr, ParsePosition pp) {
+    private static String parseLanguage(StringTokenIterator itr, ParsePosition pp) {
         if (itr.isDone() || pp.getErrorIndex() != -1) {
-            return false;
+            return EMPTY_SUBTAG;
         }
 
-        boolean found = false;
-
         String s = itr.current();
         if (isLanguage(s)) {
-            found = true;
-            language = s;
             pp.setIndex(itr.currentEnd());
             itr.next();
+            return s;
         }
 
-        return found;
+        return EMPTY_SUBTAG;
     }
 
-    private boolean parseExtlangs(StringTokenIterator itr, ParsePosition pp) {
+    private static List<String> parseExtlangs(StringTokenIterator itr, ParsePosition pp) {
         if (itr.isDone() || pp.getErrorIndex() != -1) {
-            return false;
+            return EMPTY_SUBTAGS;
         }
-
-        boolean found = false;
-
+        List<String> extLangs = null;
```

Comment:
Do we want to name this `extLangs` or `extlangs`? The parsing method and the record uses `extlangs`.

Reply:
Changed to `extlangs`, more consistent.